services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -391,4 +406,43 @@ public Object getUserInfo(UserAuthorization userAuthorization, String oauthClien
 			return new JWTWrapper(jwtIdToken);
 		}
 	}
+	
+	// If the client is not verified the message includes the URL to the synapse OAuth docs iff the user id is the same as the client creator
+	protected void validateClientVerificationStatus(String clientId, String userId) throws NotFoundException, OAuthClientNotVerifiedException {
+		validateClientVerificationStatus(clientId, () -> 
+			userId != null && oauthClientDao.getOAuthClientCreator(clientId).equals(userId)
+		);
+	}
+	
+	// If the client is not verified the message includes the URL to the synapse OAuth docs
+	protected void validateClientVerificationStatus(String clientId) throws NotFoundException, OAuthClientNotVerifiedException {
+		validateClientVerificationStatus(clientId, () -> true);
+	}
+	
+	/**
+	 * Validates that the verified flag is true for the client with the given id, the given condition is
+	 * evaluated iff the client is not verified. If the condition evaluates to true the error message
+	 * will include the URL for the synapse OAuth docs.
+	 * 
+	 * @param clientId                     The id of the client to validate
+	 * @param includeOAuthDocsURLCondition A condition that is evaluated if the client is not verified
+	 *                                     to include in the error message the URL to the OAuth docs
+	 * @throws OAuthClientNotVerifiedException If the client is not verified
+	 * @throws NotFoundException               If a client with the given id does not exist
+	 */
+	private void validateClientVerificationStatus(String clientId, BooleanSupplier includeOAuthDocsURLCondition) throws NotFoundException, OAuthClientNotVerifiedException {
+		
+		if (oauthClientDao.isOauthClientVerified(clientId)) {
+			return;
+		}
+		
+		StringBuilder errorMessage = new StringBuilder("The client is not verified yet.");
+
+		if (includeOAuthDocsURLCondition.getAsBoolean()) {
```

Comment:
I feel that the response should behave the same regardless of who the caller/user is.   In general when we try to anticipate user behavior and do something unexpected, we can end up causing trouble.  E.g., if the developer uses two Synapse accounts for testing, our check will not identify the developer.  If the developer learns to expect one error message and then gets a modified one, they may wonder why.  I think it's sufficient to say that the reason the call failed is that "The client is not verified".  The user who set up the client should have read the set-up instructions and/or will refer back to the instructions when they see this message, and the instructions will tell them what to do next.